src/classes/ACCT_IndividualAccounts_TDTM.cls

```diff
@@ -13,457 +13,454 @@
+
+            } else if (needsHHAccountRename(contactRecord, oldContact)) {
+                householdIdsToRename.add(contactRecord.AccountId);
+            }
+
+            index++;
+        }
+
+        updateOneToOneAccounts(oneToOneContactsToUpdate);
+        moveOppsToContactAccount(newHouseholdIdByContactId, oldHouseholdIdByContactId);
+        updateHHAccount(householdIdsToUpdate);
+        renameHHAccounts(householdIdsToRename);
+        updateOwners(contactWithNewOwnerById);
     }
 
     /*******************************************************************************************************
-    * @description Builds SOQL on Contact Sobject. Fields include Household Naming Settings Format fields.
-    * @param hhNamingFields Household Naming Settings Format field names
-    * @return String SOQL
-    */ 
-    @TestVisible
-    private String buildContactSoql(Set<String> hhNamingFields) {
-        List<String> selectFields = new List<String>{
-            'Id', 'npe01__Private__c', 'AccountId', 'Account.npe01__SYSTEMIsIndividual__c',
-            'Account.npe01__SYSTEM_AccountType__c', 'Account.npe01__One2OneContact__c', 'npe01__Organization_Type__c', 'Account.Name',
-            'FirstName', 'LastName', 'OwnerId', 'Salutation', 'npo02__Naming_Exclusions__c', 'npo02__Household_Naming_Order__c',
-            'MailingStreet', 'MailingCity', 'MailingState', 'MailingPostalCode', 'MailingCountry', 'MailingLatitude', 'MailingLongitude',
-            'OtherStreet', 'OtherCity', 'OtherState', 'OtherPostalCode', 'OtherCountry', 'OtherLatitude', 'OtherLongitude',
-            'Phone', 'Fax'
-        };
-        
-        if (ADDR_Addresses_TDTM.isStateCountryPicklistsEnabled) {
-            selectFields.add('MailingCountryCode');
-            selectFields.add('MailingStateCode');
-            selectFields.add('OtherCountryCode'); 
-            selectFields.add('OtherStateCode');
+    * @description Updates the accounts for contacts that have been deleted if they have other hh members
+    * @param deletedContacts the list of Contacts that need to be considered
+    * @return void
+    */
+    private void processAccountsAfterDelete(List<Contact> deletedContacts) {
+        List<Id> accountIds = new List<Id>();
+        List<Id> householdIdsToUpdate = new List<Id>();
+        List<Account> accounts = new List<Account>();
+
+        for (Contact thisContact : deletedContacts) {
+            accountIds.add(thisContact.accountId);
         }
 
-        selectFields.addAll(hhNamingFields);
+        accounts = [SELECT Id, Type, npe01__SYSTEM_AccountType__c, (SELECT Id FROM Opportunities LIMIT 1), (SELECT Id FROM Contacts LIMIT 1) FROM Account WHERE Id IN :accountIds];
 
-        return new UTIL_Query()
-            .withSelectFields(selectFields)
-            .withFrom('Contact')
-            .withWhere('Id IN :contacts')
-            .build();
+        for (Account accountRecord : accounts) {
```

Comment:
Could the query just be part of the for loop rather than query in the list first?